src/_guides/language/effective-dart/usage_migrated.md

```diff
@@ -321,6 +321,27 @@ int measureMessage(String? message) {
 {% endprettify %}
 
 
+### AVOID `late` variables if you need to check whether they are initialized.
+
+Dart offers no way to tell if a `late` variable
+has been initialized or assigned to.
+If you access it, it either immediately runs the initializer
+if it has one or throws an exception.
+Sometimes you have some state that is lazily initialized
+where `late` would be a good fit,
+but you also need to be able to *tell* if the initialization has happened yet.
+
+You could do that by storing the state in a `late` variable
+and then have a separate boolean field
+to track whether the variable has been set.
+This works, but is redundant since Dart *internally*
+still needs to maintain the initialized status of the `late` variable.
```

Comment:
```suggestion
Although you could detect initialization by storing the state in a `late` variable
and having a separate boolean field
that tracks whether the variable has been set,
that's redundant because Dart *internally*
maintains the initialized status of the `late` variable.
```